### PR TITLE
pm2: init at 4.5.0

### PR DESCRIPTION
**Pull request by jeremyschlatter**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/node-packages/node-env.nix [hunk 1]
```diff
@@ -372,37 +372,8 @@ let
...
-      if builtins.isAttrs ff then (ff // {
-        overrideNodeAttrs = newArgs: makeOverridableNodePackage f (overrideWith newArgs);
-      })
-      else if builtins.isFunction ff then {
-        overrideNodeAttrs = newArgs: makeOverridableNodePackage f (overrideWith newArgs);
-        __functor = self: ff;
-      }
-      else ff;
-
-
```
Comment (jeremyschlatter) [inline pkgs/development/node-packages/node-env.nix:403]: I'm not sure what's up with `makeOverridableNodePackage`. This was deleted by `generate.sh`, but I'm not sure if it should have been. It was implemented very recently.

@jtojnar is this deletion ok? Should `generate.sh` be preserving this function?

**Comment by aanderse** [comment]
See https://github.com/NixOS/nixpkgs/pull/86840 and https://github.com/NixOS/nixpkgs/pull/99617.

File: pkgs/development/node-packages/node-env.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/node-packages/node-env.nix:403]: It should not be deleted, it is required for `expo-cli`. But it seems to be possible to do without: https://github.com/NixOS/nixpkgs/pull/100015

Comment (jeremyschlatter) [reply pkgs/development/node-packages/node-env.nix]: Thanks. I just rebased this PR with #100015 included, so this diff is now gone.

**Comment by jeremyschlatter** [comment]
Whoops, I didn't notice those other pm2 PRs when I opened this. Thanks for linking them.

Tempted to close this one, but I'll leave it open for now in case it's easier to merge. (The other two currently have merge conflicts.)

**Comment by jeremyschlatter** [comment]
@jtojnar Would you be up for reviewing this?

I'm still learning how the review process works around here. 😅  I've had a couple of packages merged so far, but the people who merged them showed up by magic and I don't know who to ask for help when that doesn't happen.

**Comment by jtojnar** [comment]
There is not really much to review. Let’s just merge this before there are conflicts again.

**Comment by adminy** [comment]
```
root in ~/home on master ● ● λ pm2 startup
[PM2] Init System found: systemd
Platform systemd
Template
[Unit]
Description=PM2 process manager
Documentation=https://pm2.keymetrics.io/
After=network.target

[Service]
Type=forking
User=root
LimitNOFILE=infinity
LimitNPROC=infinity
LimitCORE=infinity
Environment=PATH=/run/wrappers/bin:/root/.nix-profile/bin:/etc/profiles/per-user/root/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/nix/store/pb6a5bb4z4i8y8r85kmxqq3q7h0si9pw-nodejs-18.16.1/bin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
Environment=PM2_HOME=/root/.pm2
PIDFile=/root/.pm2/pm2.pid
Restart=on-failure

ExecStart=/nix/store/kp4jprgxfkr2byxnav6didr8lmypvj65-pm2-5.3.0/lib/node_modules/pm2/bin/pm2 resurrect
ExecReload=/nix/store/kp4jprgxfkr2byxnav6didr8lmypvj65-pm2-5.3.0/lib/node_modules/pm2/bin/pm2 reload all
ExecStop=/nix/store/kp4jprgxfkr2byxnav6didr8lmypvj65-pm2-5.3.0/lib/node_modules/pm2/bin/pm2 kill

[Install]
WantedBy=multi-user.target

Target path
/etc/systemd/system/pm2-root.service
Command list
[ 'systemctl enable pm2-root' ]
[PM2] Writing init configuration in /etc/systemd/system/pm2-root.service
```
Error:
```
[PM2][ERROR] Failure when trying to write startup script
EROFS: read-only file system, open '/etc/systemd/system/pm2-root.service'
root in ~/home on master ● ● λ 
```

**Pull request merged**